src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs

```diff
@@ -241,182 +243,200 @@ public async Task<IActionResult> StandardDetails(string referenceNumber)
-        [HttpPost("standard/{id}/confirm-standard/{standardReference}/{version}")]
-        public async Task<IActionResult> ConfirmStandard(StandardVersionViewModel model, Guid id, string standardReference, string version)
+        [PrivilegeAuthorize(Privileges.ApplyForStandard)]
+        [HttpGet("standard/{id}/apply-standard/{search}/{referenceNumber}/confirm", Name = ApplyStandardConfirmRouteGet)]
+        [ApplicationAuthorize(routeId: "Id")]
+        [ModelStatePersist(ModelStatePersist.RestoreEntry)]
+        public async Task<IActionResult> ApplyStandardConfirm(Guid id, string search, string referenceNumber)
         {
+            if (id == Guid.Empty)
+                throw new ArgumentException($"Value of {nameof(id)} cannot be empty");
+
+            if (string.IsNullOrEmpty(search))
+                throw new ArgumentException($"Value of {nameof(search)} cannot be null or empty");
+
+            if (string.IsNullOrEmpty(referenceNumber))
+                throw new ArgumentException($"Value of {nameof(referenceNumber)} cannot be null or empty");
+
             var application = await _applicationApiClient.GetApplication(id);
             if (!CanUpdateApplicationAsync(application))
             {
                 return RedirectToAction("Applications", "Application");
             }
 
             var org = await _orgApiClient.GetEpaOrganisation(application.OrganisationId.ToString());
-            var standardVersions = (await _orgApiClient.GetAppliedStandardVersionsForEPAO(org?.OrganisationId, standardReference))
-                                        .OrderBy(s => s.Version);
-
-            bool anyExistingVersions = standardVersions.Any(x => x.ApprovedStatus == ApprovedStatus.Approved || x.ApplicationStatus == ApplicationStatus.Submitted);
-
-            AppliedStandardVersion selectedStandard = null;
-            string applicationStatus = null;
-            List<string> versions = null;
+            var standardVersions = await _orgApiClient.GetAppliedStandardVersionsForEPAO(org?.OrganisationId, referenceNumber);
+            var earliestStandard = standardVersions.OrderBy(s => s.Version).FirstOrDefault();
+            var latestStandard = standardVersions.OrderBy(s => s.Version).LastOrDefault();
 
-            if (string.IsNullOrWhiteSpace(version))
+            if (standardVersions.Any(x => x.ApprovedStatus == ApprovedStatus.Approved))
             {
-                selectedStandard = standardVersions.LastOrDefault();
-                versions = model.SelectedVersions ?? new List<string> { selectedStandard.Version };
-                if (model.SelectedVersions != null)
-                    applicationStatus = await ApplicationStandardStatus(application, standardReference, model.SelectedVersions);
+                return RedirectToRoute(StandardDetailsRouteGet, new { referenceNumber });
             }
-            else
+            else if(latestStandard.EqaProviderName == "Ofqual")
```

Comment:
We should account for latestStandard being null here

Reply:
Added handling for various cases